### DISCUSSION TITLE
Where is the documentation/API reference of the task decorator (@task)?

### DISCUSSION THREAD

**Discussion by scimas** [apache/airflow#30686] [Q&A]
The [task flow](https://airflow.apache.org/docs/apache-airflow/2.4.3/concepts/taskflow.html) page refers this `airflow.decorators` module, but there is no such module listed in the [python API](https://airflow.apache.org/docs/apache-airflow/2.4.3/python-api-ref.html). So I have no idea where to look for which arguments can(not) be passed to things like `@task` and `@task.sensor`.

**Comment by potiuk** [comment]
I don't think there is one. This is not a typical object/class - so it has no "docstring" with "parameters" because it is considerably more complex to implement.  The vaild parameters should usually come as autocomplete in the IDE and you can also look at the taskflow tutorial and https://github.com/apache/airflow/blob/main/airflow/decorators/__init__.pyi

**Reply by scimas** [reply]
If documentation cannot be provided, why is it the [recommended](https://airflow.apache.org/docs/apache-airflow/2.4.3/upgrading-from-1-10/index.html#migrating-to-taskflow-api) way to work with python tasks?
![image](https://user-images.githubusercontent.com/16473290/232568015-cedc6322-e7f9-43fd-a2d0-c591817bdf8f.png)

Looking at that how to guide, it seems like `@task` supports the same arguments as the `PythonOperator`. But then there's something call `multiple_outputs` which is only supported by `@task`, and the only mentions of it I can find is on the task flow API tutorial page. So I'm left wondering if there are other such arguments that are not listed anywhere.

IDEs don't seem to help, I have tried using both visual studio code with the python/pylance plugin and Pycharm. Both simply show that `@task`'s type is `TaskDecoratorCollection` and nothing else.

And thank you for linking to the stub file! Needing to go into airflow's code for such things is not really productive though.

**Reply by potiuk** [reply]
Agree - it could be better described. And invite you to help is improve it if you think going to the stub file is not good for productivity

I think this is absolutely no problem to add a documentation about this at the Python page you descrube . If you think that others can be helped by it - feel free to add some more documentation about what you've just learned. Airflow is created by almost 2500 contributors, volunteers, most of them people like you who missed something in the docs and added it. It's actually quite easy  - you can go to the Python operator's doc page and add a doc about what you've just lerned by clicking "Suggest a change on this page" - this will open a PR and you will be able to describe it in the way you .

Why not you read the stub I provided, and write the doc in the  way you understand it and where you will describe it in the way that you think will be best for people like you? Then, when you open the PR, some committers migh comment on your findings and suggest improvements. If your goal is to have that documentation, this is the fastest and most productive way to make it happen - also you will learn a lot about it by doing it.

Why not?
